server/api/services/repo.service.js

```diff
@@ -132,25 +132,25 @@ const deleteRepo = async ({ repoName, username }) => {
     await deleteByUserAndReponame({ owner: username, reponame: repoName });
     return true;
   } catch (e) {
-    return false;
+    return e;
   }
 };
 
-const getReposNames = async ({ user, filter, limit }) => {
-  const pathToRepo = repoHelper.getPathToRepos(user);
-  const doesDirExists = fs.existsSync(pathToRepo);
-  if (doesDirExists) {
-    const repos = fs
-      .readdirSync(pathToRepo, { withFileTypes: true })
-      .filter(dir => dir.isDirectory())
-      .map(dir => dir.name.slice(0, -4));
-    const filteredRepos = filter ? repos.filter(repo => repo.includes(filter)) : repos;
-    return limit ? filteredRepos.slice(0, limit) : repos;
-  }
-  return [];
+const getReposNames = async ({ user: username, filter, limit }) => {
+  const { id } = await userRepository.getByUsername(username);
+  const findOptions = {
+    filter,
+    limit,
+    sortByCreatedDateDesc: true
```

Comment:
Added auto sorting by latest repos on my own. If we don't need that, I'll remove :)